# Dataset Selection: Used Car Price Prediction
**Objective:** Profile raw data sources to select the primary baseline dataset for the price prediction model.

In [1]:
import pandas as pd

# 1. Load the two datasets
df_v1 = pd.read_csv('../data/raw/used_cars_dataset_v1.csv')
df_v2 = pd.read_csv('../data/raw/used_cars_dataset_v2.csv')


def profile_dataset(df, name):
    print(f"--- Profiling {name} ---")
    # Get the number of rows and columns
    print(f"Shape (Rows, Columns): {df.shape}")
    
    # Get the list of column names
    print(f"Columns: {list(df.columns)}")
    
    # Count exact duplicate rows
    print(f"Duplicate Rows: {df.duplicated().sum()}")
    
    # Count missing (NaN) values per column, only showing columns with > 0 missing
    missing_counts = df.isna().sum()
    missing_filtered = missing_counts[missing_counts > 0]
    
    if len(missing_filtered) > 0:
        print("Missing Values:\n", missing_filtered.to_string())
    else:
        print("Missing Values: None")
    print("\n")


# 2. Run the profiler on both dataframes
profile_dataset(df_v1, "Dataset V1")
profile_dataset(df_v2, "Dataset V2")

--- Profiling Dataset V1 ---
Shape (Rows, Columns): (9582, 11)
Columns: ['Brand', 'model', 'Year', 'Age', 'kmDriven', 'Transmission', 'Owner', 'FuelType', 'PostedDate', 'AdditionInfo', 'AskPrice']
Duplicate Rows: 724
Missing Values:
 kmDriven    47


--- Profiling Dataset V2 ---
Shape (Rows, Columns): (14993, 11)
Columns: ['Brand', 'model', 'Year', 'Age', 'kmDriven', 'Transmission', 'Owner', 'FuelType', 'PostedDate', 'AdditionInfo', 'AskPrice']
Duplicate Rows: 997
Missing Values:
 kmDriven    88




## Decision: `used_cars_dataset_v2.csv`

**Justification:**
* **Feature space:** Identical to v1.
* **Volume:** Yields ~14k unique records vs v1's ~8.8k.
* **Completeness:** Nulls are negligible (0.5%, isolated to `kmDriven`).

**Pipeline Requirements (Next Steps):**
* Drop 997 exact duplicates prior to train/test split to prevent target leakage.
* Resolve 88 nulls in `kmDriven`.